app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/CheckAllAnswersAndSubmitController.scala

```diff
@@ -72,33 +73,63 @@ class CheckAllAnswersAndSubmitController @Inject() (
           )
         val result =
           for {
-            response <- returnsService.submitReturn(submittedReturnRequest)
+            response <- returnsService
+                         .submitReturn(submittedReturnRequest)
+                         .map(Right(_))
+                         .leftFlatMap[Either[Error, SubmitReturnResponse], Error](e => EitherT.pure(Left(e)))
+            newJourneyStatus = response.fold(
+              _ =>
+                SubmitReturnFailed(
+                  fillingOutReturn.subscribedDetails,
+                  fillingOutReturn.ggCredId,
+                  fillingOutReturn.agentReferenceNumber
+                ),
+              submitReturnResponse =>
+                JustSubmittedReturn(
+                  fillingOutReturn.subscribedDetails,
+                  fillingOutReturn.ggCredId,
+                  fillingOutReturn.agentReferenceNumber,
+                  completeReturn,
+                  submitReturnResponse
+                )
+            )
             _ <- EitherT(
                   updateSession(sessionStore, request)(
-                    _.copy(journeyStatus = Some(
-                      JustSubmittedReturn(
-                        fillingOutReturn.subscribedDetails,
-                        fillingOutReturn.ggCredId,
-                        fillingOutReturn.agentReferenceNumber,
-                        completeReturn,
-                        response
-                      )
-                    )
-                    )
+                    _.copy(journeyStatus = Some(newJourneyStatus))
                   )
                 )
-          } yield ()
+          } yield response
 
         result.fold(
           { e =>
-            logger.warn("Error while try to submit return and udpate session", e)
+            logger.warn("Error while trying to update session", e)
```

Comment:
`result` could fail with failed submission, so this error message is not really reflecting that